plc4j/tools/connection-cache/src/main/java/org/apache/plc4x/java/utils/cache/ConnectionContainer.java

```diff
@@ -28,12 +28,20 @@
 import java.util.concurrent.Future;
 
 class ConnectionContainer {
-    private final PlcConnection connection;
+    private PlcConnection connection;
+    private boolean closed = false;
     private final Duration maxLeaseTime;
     private final Queue<CompletableFuture<PlcConnection>> queue;
 
     private LeasedPlcConnection leasedConnection;
 
+    public boolean isClosed() {
+        return closed;
+    }
+    public PlcConnection getRawConnection() {
```

Comment:
In contrast to your comment, this does actually return the Plc4x connection and not the Netty connection ... we shouldn't expose this, as this allows "stealing" the connection.